service/deployment-guard/main.go

```diff
@@ -36,6 +36,21 @@ func main() {
 	}
 	ctx := context.Background()
 
+	// NOTE: when we are running alongside odf-operator the CRDs corresponding
+	// to an operator is installed first however it isn't guaranteed that CRDs
+	// of all dependent operators are installed first before deploying any CSV.
+	//
+	// Due to above, StorageCluster CRD is sometimes created after we check for
+	// it's existence and 2 minutes is chosen as it is double the time observed
+	// when client-op is listed first and ocs-op listed last in the install plan
+	// for deploying CSVs.
+	//
+	// Please note this should be temporary upto when we develop new operator
+	// which directly installs other operators based on runtime requirements
+	// outside of OLM.
+	klog.Info("Waiting for 2 minutes before checking to allow operator to run")
+	time.Sleep(2 * time.Minute)
```

Comment:
why not increase the Cap instead of putting a hard sleep of 2 minutes?

Reply:
that'll not work, there are two branches, the backoff is for what happens after we found the existence of CRD, the hard sleep is waiting to check for the CRD creation.

Reply:
IMO hard sleep might work in all cases but again it can cause issues as well if the timing is not met (in worst cases), can we check for CRD existing as well allowOperatorToRun function?

Reply:
got to know that we will not have CRD when running in consumer mode, resolving now